Documentation/contributing/testing/e2e.rst

```diff
@@ -166,3 +165,45 @@ To stop the VM, run from the host:
 .. code-block:: shell-session
 
     $ pkill qemu-system-x86
+
+Running tests in VM with custom kernel
+""""""""""""""""""""""""""""""""""""""
+
+It is possible to test Cilium on a LVH VM with a custom built Linux kernel (e.g.,
+for fast testing iterations when doing kernel development work for Cilium features).
+
+First, to configure and to build the kernel:
+
+.. code-block:: shell-session
+
+   $ git clone --depth=1 https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git
+   $ cd bpf-next/ # hack hack hack
+
+   # configure kernel, so that it can be run in LVH VM:
+   $ git clone https://github.com/cilium/little-vm-helper-images
+   $ cat ../little-vm-helper-images/_data/kernels.json | \
+        jq -r '.common_opts.[] | (.[0])+" "+(.[1])' | \
+        xargs ./scripts/config
+
+   $ make -j$(nproc)
+
+Second, start the LVH VM with the custom kernel:
+
+.. code-block:: shell-session
+
+   $ lvh run --image ./images/kind_bpf-next.qcow2 \
+        --host-mount $(pwd) \
+        --kernel ./bpf-next/arch/x86_64/boot/bzImage \
+        --daemonize -p 2222:22 --cpu=3 --mem=6G \
+
+Third, SSH into VM, and install the custom kernel modules (the step is no longer
+required once `little-vm-helper#117 <https://github.com/cilium/little-vm-helper/issues/117>`_
+has been resolved):
+
+.. code-block:: shell-session
+
+    $ ssh -p 2222 -o "StrictHostKeyChecking=no" root@localhost
+    # cd /host/bpf-next
+    # make modules_install
+
+Finally, you can use instructions from the previous chapter to run and to test Cilium.
```

Comment:
Could you link the relevant section instead of saying "the previous chapter"?

Reply:
Aye, good idea! Let me figure out how to link a chapter in RST :sweat_smile: